dynamiqs/hermitian_conjugate.py

```diff
@@ -0,0 +1,23 @@
+from jaxtyping import ArrayLike
+from qutip import Qobj
+
+from .qarrays import QArray, QArrayLike, asqarray
+
+
+class HermitianConjugate:
+    def __radd__(self, y: QArrayLike) -> QArrayLike:
+        if isinstance(y, Qobj):
+            y = asqarray(y)
```

Comment:
The input object should not be converted silently (or at least, be converted when it's a Qobj but not when it's an ArrayLike). Either we return a `Qobj` or we don't support `Qobj` at all IMO.